### PR TITLE
added isadmin property to getCurrentUser call so we can hide certain …

### PR DESCRIPTION
**Pull request by ARADDCC002**
…pages from nonadmins

Note this is used for client-side validation ONLY. This should not be used for checking for anything important - it's main purpose is to hide pages/sections that normal users are currently not able to use anyway.

### CODE REVIEW

File: frontend/pages/schemas/list.tsx [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
 import SchemaTab from 'src/schemas/SchemaTab'
 
 export default function SchemasPage() {
+  const { currentUser, isCurrentUserLoading, isCurrentUserError } = useGetCurrentUser()
+
+  if (isCurrentUserLoading) return <Loading />
+
+  if (!currentUser || !currentUser.isAdmin) {
+    return (
+      <Forbidden errorMessage='If you think this is in error please contact the model owners.' noMargin hideNavButton />
```
Comment (a3957273) [inline frontend/pages/schemas/list.tsx:18]: Should say "Bailo admins" or equivalent, not model owners.

File: frontend/pages/schemas/list.tsx [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+    )
+  }
+
+  const error = MultipleErrorWrapper(`Unable to load model page`, {
```
Comment (a3957273) [inline frontend/pages/schemas/list.tsx:22]: This is not a model page.

File: frontend/types/types.ts [hunk 3]
```diff
@@ -161,6 +161,7 @@ export interface PostSimpleUpload {
 
 export interface User {
   dn: string
+  isAdmin?: boolean
```
Comment (a3957273) [inline frontend/types/types.ts:164]: Why is this optional?

File: backend/src/routes/v2/entities/getCurrentUser.ts [hunk 4]
```diff
@@ -34,7 +36,6 @@ export const getCurrentUser = [
   bodyParser.json(),
   async (req: Request, res: Response<GetCurrentUserResponses>) => {
     const _ = parse(req, getCurrentUserSchema)
-
-    return res.json({ user: req.user })
+    return res.json({ user: { ...req.user, isAdmin: await authentication.hasRole(req.user, Roles.Admin) } })
```
Comment (a3957273) [inline backend/src/routes/v2/entities/getCurrentUser.ts:39]:
```typescript
const _ = parse(req, getCurrentUserSchema)
const isAdmin = await authentication.hasRole(req.user, Roles.Admin)
return res.json({ user: { ...req.user, isAdmin }})
```

Splitting it out onto a separate line improves the readability, especially if we add more values.

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: frontend/src/wrapper/SideNavigation.tsx [hunk 5]
```diff
@@ -152,7 +152,7 @@ export default function SideNavigation({
             />
             <Divider />
             {/* TODO Once currentUser api has been updated to use roles we should check if they're admin */}
```
Comment (ARADDCC012) [inline frontend/src/wrapper/SideNavigation.tsx:154]: This comment can be removed now

**Review by ARADDCC012 — CHANGES_REQUESTED** [review]

**Review by a3957273 — APPROVED** [review]

**Pull request merged**